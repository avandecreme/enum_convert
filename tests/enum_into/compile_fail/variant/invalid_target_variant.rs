use enum_convert::EnumInto;

#[derive(EnumInto)]
#[enum_into(Target)]
enum Source {
    #[enum_into(Target::NonExistent)] // Invalid variant name
    Unit,
}

enum Target {
    Unit,
}

fn main() {}
