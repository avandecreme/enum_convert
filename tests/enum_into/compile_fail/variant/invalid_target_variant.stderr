error[E0599]: no variant or associated item named `NonExistent` found for enum `Target` in the current scope
  --> tests/enum_into/compile_fail/variant/invalid_target_variant.rs:6:25
   |
6  |     #[enum_into(Target::NonExistent)] // Invalid variant name
   |                         ^^^^^^^^^^^ variant or associated item not found in `Target`
...
10 | enum Target {
   | ----------- variant or associated item `NonExistent` not found for this enum
